### PR TITLE
S3 file storage

### PR DESCRIPTION
**Pull request by silveirado**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,14 +22,26 @@
 -   `DEFAULT_SMTP_TLS_REJECT_UNAUTHORIZED`: SMTP config `tls.rejectUnauthorized` for nodemailer, config would open a connection to TLS server with self-signed or invalid TLS certificate
 -   `DEFAULT_SMTP_AUTH_METHOD`: SMTP config `authMethod` for nodemailer, defines preferred authentication method, defaults to ‘PLAIN’
 -   `DEFAULT_SMTP_DEBUG`: SMTP config `debug` for nodemailer, if set to true, then logs SMTP traffic, otherwise logs only transaction events
+-   `UI_URL`: host for ui
 
 ## FILE STORAGE API
 
+-   `STORAGE`: Can be `s3` or `fs` for files and images uploads
```
Comment (marcusdemoura) [inline README.md:29]: `STORAGE`: (optional) Can be `s3` or `fs` for files and images uploads

File: README.md [hunk 2]
```diff
@@ -23,14 +22,26 @@
...
+-   `BLOB_URL`: (optional) if use external server for file upload
```
Comment (marcusdemoura) [inline README.md:30]: `BLOB_URL`: (optional) if use external server for file upload. if `STORAGE` is `s3` you must set a BLOB_URL

File: README.md [hunk 3]
```diff
@@ -23,14 +22,26 @@
...
+-   `PREVIEW_URL`: (optional) if use external file server
```
Comment (marcusdemoura) [inline README.md:31]: `PREVIEW_URL`: (optional) if use external file server. if `STORAGE` is `fs` you must set a PREVIEW_URL

File: README.md (hunk 1 above)
Comment (silveirado) [reply README.md:29]: This is optional if you use an external server for file storage.

File: README.md (hunk 3 above)
Comment (marcusdemoura) [inline README.md:31]: `BLOB_URL`: (optional) if use external server for file upload. if `STORAGE` is `s3` you must set a BLOB_URL
`PREVIEW_URL`: (optional) if use external file server. if `STORAGE` is `fs` you must set a PREVIEW_URL

Comment (silveirado) [reply README.md:31]: `BLOB_URL` and `PREVIW_URL` are required only if you have an external file server.

File: README.md [hunk 4]
```diff
@@ -23,14 +22,26 @@
...
+
+### s3 STORAGE SETTINGS
```
Comment (marcusdemoura) [inline README.md:33]: required if `STORAGE` set to `s3`

File: README.md [hunk 5]
```diff
@@ -23,14 +22,26 @@
...
+
 -   `S3_DOMAIN`: required if different of AWS eg: `digitaloceanspaces.com`
 -   `S3_REGION`: S3 region
 -   `S3_BUCKET`: S3 bucket
 -   `S3_ACCESSKEY`: Generated for your aws account. Follow this instructions: [Where’s My Secret Access Key?](https://aws.amazon.com/blogs/security/wheres-my-secret-access-key/).
 -   `S3_SECREDKEY`: Generate with instructions above (👆).
+-   `S3_PUBLIC_URL`: Bucket public url
+
+### fs STORAGE SETTINGS
```
Comment (marcusdemoura) [inline README.md:42]: required if `STORAGE` set to `fs`

File: README.md (hunk 1 above)
Comment (marcusdemoura) [reply README.md:29]: `STORAGE`: (optional) if use external server for files and images storage. Can be `s3` or `fs`.

File: server/routes/rest/view/view.js [hunk 6]
```diff
@@ -78,16 +79,11 @@ app.get('/', function(req, res, next) {
 		locale: user.locale,
 		lbl_loading: 'Carregando o sistema...',
 		btn_close: 'Fechar',
-		timeInMilis: +new Date()
+		timeInMilis: +new Date(),
+		uiServer: process.env.UI_URL || 'ui.konecty.com',
+		blobUrl: process.env.BLOB_URL == null ? '' : `//${process.env.BLOB_URL}`,
```
[the remark this answers is not in the document]
Comment (silveirado) [reply server/routes/rest/view/view.js]: We'll extract the hostname from parameter with or without the protocol.

**Pull request merged**